src/components/sbb-calendar/sbb-calendar.tsx

```diff
@@ -822,209 +827,207 @@ export class SbbCalendar implements ComponentInterface {
-        return (
+        return html`
           <tr>
-            {[...Array(firstRowOffset).keys()].map(() => (
-              <td
-                class="sbb-calendar__table-data"
-                data-day={`0 ${week[0].monthValue} ${week[0].yearValue}`}
-              ></td>
-            ))}
-            {this._createDayCells(week, today)}
+            ${[...Array(firstRowOffset).keys()].map(
+              () =>
+                html`<td
+                  class="sbb-calendar__table-data"
+                  data-day=${`0 ${week[0].monthValue} ${week[0].yearValue}`}
+                ></td>`,
+            )}
+            ${this._createDayCells(week, today)}
           </tr>
-        );
+        `;
       }
-      return <tr>{this._createDayCells(week, today)}</tr>;
+      return html`<tr>
+        ${this._createDayCells(week, today)}
+      </tr>`;
     });
   }
 
   /** Creates the cells for the daily view. */
-  private _createDayCells(week: Day[], today: string): JSX.Element {
+  private _createDayCells(week: Day[], today: string): TemplateResult[] {
     return week.map((day: Day) => {
       const isOutOfRange = !this._isDayInRange(day.value);
-      const isFilteredOut = !this.dateFilter(this._dateAdapter.createDateFromISOString(day.value));
+      const isFilteredOut = !this?.dateFilter(this._dateAdapter.createDateFromISOString(day.value));
       const selected: boolean = !!this._selected && day.value === this._selected;
       const dayValue = `${day.dayValue} ${day.monthValue} ${day.yearValue}`;
       const isToday = day.value === today;
-      return (
+      return html`
         <td
-          class={{
+          class=${classMap({
             'sbb-calendar__table-data': true,
             'sbb-calendar__table-data-selected': selected,
-          }}
+          })}
         >
           <button
-            class={{
+            class=${classMap({
               'sbb-calendar__cell': true,
               'sbb-calendar__day': true,
               'sbb-calendar__cell-current': isToday,
               'sbb-calendar__selected': selected,
               'sbb-calendar__crossed-out': !isOutOfRange && isFilteredOut,
-            }}
-            onClick={() => this._selectDate(day.value)}
-            disabled={isOutOfRange || isFilteredOut}
-            aria-label={this._dateAdapter.getAccessibilityFormatDate(day.value)}
-            aria-pressed={String(selected)}
-            aria-disabled={String(isOutOfRange || isFilteredOut)}
-            aria-current={isToday ? 'date' : undefined}
-            data-day={dayValue}
+            })}
+            @click=${() => this._selectDate(day.value)}
+            ?disabled=${isOutOfRange || isFilteredOut}
+            aria-label=${this._dateAdapter.getAccessibilityFormatDate(day.value)}
+            aria-pressed=${selected}
+            aria-disabled=${isOutOfRange || isFilteredOut}
+            aria-current=${isToday ? 'date' : nothing}
+            data-day=${dayValue}
```

Comment:
Not 100% sure that the suggestion makes sense
```suggestion
            data-day=${dayValue || nothing}
```